app/components/common/Header.js

```diff
@@ -1,31 +1,43 @@
 import React, { PropTypes } from 'react';
 import { Link, IndexLink } from 'react-router';
 
-const Header = () => {
+const Header = (props) => {
+  let links = [
+    { to: '/app/about', className: 'glyphicon glyphicon-exclamation-sign', label: 'About'}
+  ];
+  if(props.showAllLinks) {
+    links.unshift(
+      { to: '/app/users', className: 'glyphicon glyphicon-user', label: 'Users'}
+    );
+    links.unshift(
+      { to: '/', className: 'glyphicon glyphicon-home', label: 'Home'}
+    );
+  }
+
+  const linksSection = links.map((link, index) => {
+    return (
+      <li key={index} >
+        <Link to={link.to} activeClassName="active"><i className={`${link.className}`}/> {link.label}</Link>
+      </li>
+    );
+  });
```

Comment:
Super-minor: since this is a single statement lambda, you can avoid the return by using parenthesis and forcing it into an expression:

```javascript
const linksSection = links.map((link, index) => (
    <li key={index}>
    // ...
));
```